autorest/codegen/serializers/__init__.py

```diff
@@ -150,6 +167,11 @@ def _serialize_and_write_package_files_proc(**kwargs: Any):
                 ):
                     template = env.get_template(template_name)
                     render_result = template.render(**kwargs)
+                    # make sure some files will be formatted(setup.py, etc)
+                    if ".py" in file:
+                        render_result = black.format_file_contents(
+                            render_result, fast=True, mode=_BLACK_MODE
+                        )
```

Comment:
we need these code since `--black` will only format code under `output-folder`. If `output-folder` is like https://github.com/Azure/autorest.python/pull/1291/files#r893063669, `setup.py` will not be formatted.

Reply:
I think instead of adding all of this code here, we can add code to the BlackPlugin itself that checks if there's a setup.py outside of the output folder

Reply:
@iscai-msft make sense and I move the code [here](https://github.com/Azure/autorest.python/pull/1291/files#r895342694)